virtnbdbackup

```diff
@@ -59,6 +59,9 @@ def main():
     parser.add_argument(
         "-o", "--output", required=True, type=str,
         help="Output target directory")
+    parser.add_argument(
+        '-C', '--checkpointdir', required=True, type=str,
```

Comment:
> If backing up multiple virtual machines you need one checkpoint directory per virtual machine,

Yes

>  bcs the checkpoint name is quite hardcoded now. Wouldnt it make sense to store the checkpoint xml in the same  directory as the backup files then? 

Let's consider scenario:

1. Have full and incremental backups in directory /backup/domain.20210502
2. Domain stopped and migrated to another host
3. Want to make full backup to /backup/domain.20210503

We'll need to specify "previous backup directory" to access saved checkpoint configs
It is possible but will add unnecessary complexity to backup scripts using virtnbdbackup IMHO

> Otherwise, maybe it makes sense to have the virtual machines name included in the checkpoint name, but that would break existing backup chains.

Oops forgot to mention I've assumed that `checkpointdir` contains only one domain checkpoints

For example:
```
root@d02l:/backup/virtnbdbackup# virtnbdbackup -d vm-template -l inc -o /backup/vm-template.20210503 -C /var/lib/libvirt/images.drbd0/checkpoints/vm-template
2021-05-03 15:17:54 INFO common - printVersion: Version: 0.19 Arguments: /usr/local/bin/virtnbdbackup -d vm-template -l inc -o /backup/vm-template.20210503 -C /var/lib/libvirt/images.drbd0/checkpoints/vm-template
2021-05-03 15:17:54 INFO virtnbdbackup - main: Domain has 1 disks attached which support changed block tracking.
2021-05-03 15:17:54 INFO virtnbdbackup - main: Looking for checkpoints
2021-05-03 15:17:54 INFO virtnbdbackup - main: Found checkpoints, next checkpoint id: 2
2021-05-03 15:17:54 INFO virtnbdbackup - main: Parent checkpoint name virtnbdbackup.1
2021-05-03 15:17:54 INFO virtnbdbackup - main: Using checkpoint name: virtnbdbackup.2
2021-05-03 15:17:54 INFO virtnbdbackup - main: Temporary scratch file target directory: /var/tmp
2021-05-03 15:17:54 INFO virtnbdbackup - main: Starting backup job.
2021-05-03 15:17:55 INFO libvirthelper - fsFreeze: Freezed filesystems.
2021-05-03 15:17:55 INFO libvirthelper - fsThaw: Thawed filesystems.
2021-05-03 15:17:55 INFO virtnbdbackup - main: Started backup job with checkpoint, saving information.
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: INC backup: set context to qemu:dirty-bitmap:backup-vda
2021-05-03 15:17:55 INFO nbdhelper - version: 1.6.3
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: NDB Endpoint socket: /var/tmp/virtnbdbackup.963123
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: Using nbd to query extents
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: Got 317 extents
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: 42949672960 bytes disk size
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: 35782656 bytes of data extents to backup
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: Write data to target file: /backup/vm-template.20210503/vda.inc.virtnbdbackup.2.data.partial
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: Creating thin provisioned stream backup image
saving disk vda: 100%|█████████████████████████████████████████████████████████████████████| 35.8M/35.8M [00:00<00:00, 212MB/s]
2021-05-03 15:17:55 INFO virtnbdbackup - backupDisk: Backup of disk vda finished, file: /backup/vm-template.20210503/vda.inc.virtnbdbackup.2.data
2021-05-03 15:17:55 INFO virtnbdbackup - backupConfig: Saving VM config to: /backup/vm-template.20210503/vmconfig.virtnbdbackup.2.xml
2021-05-03 15:17:55 INFO virtnbdbackup - main: Stopping backup task
2021-05-03 15:17:55 INFO virtnbdbackup - main: Finished
root@d02l:/backup/virtnbdbackup# ls -l /var/lib/libvirt/images.drbd0/checkpoints/vm-template
total 36
-rw-r--r-- 1 root root 7843 May  3 12:42 virtnbdbackup.0.xml
-rw-r--r-- 1 root root 7899 May  3 12:43 virtnbdbackup.1.xml
-rw-r--r-- 1 root root 7899 May  3 15:17 virtnbdbackup.2.xml
root@d02l:/backup/virtnbdbackup#
```
with VM name already present in storage path, add one more VM name to checkpoint woud be a tautology

Reply:
> 1. Have full and incremental backups in directory /backup/domain.20210502
> 2. Domain stopped and migrated to another host
> 3. Want to make full backup to /backup/domain.20210503
> 
> We'll need to specify "previous backup directory" to access saved checkpoint configs

yes, thats true.. it needs to redefine the checkpoints if not existant so it can remove them correctly before the next full backup execution.. to have libvirt and qcow files correctly in sync.